### DISCUSSION TITLE
Support editing Amazon Managed Prometheus alertmanager rules

### DISCUSSION THREAD

**Discussion by robbierolin** [grafana/grafana#64781] [Alerting]
Today it is possible to edit external alertmanager rules from Grafana if the external alertmanager is Cortex-compatible. Amazon Managed Prometheus exposes Prometheus-compatible APIs but also exposes APIs to edit alertmanager rules through the AWS SDK ([link](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-APIReference.html#AMP-APIReference-PutAlertManagerDefinition)). Grafana should support editing alert rules in Amazon Managed Prometheus by calling these APIs using the SigV4 authentication configured for the data source. Amazon Managed Prometheus can be added as an additional `Implementation` in the configuration for the alertmanager data source.

I am interested in working on this and looking to understand if Grafana would be open to this change or have any thoughts related to the implementation.

**Comment by alexweav** [comment]
Hello! :wave: I had a couple questions, to get some clarification:

Firstly, based on this and other issues, I'm understanding there to be two separate requests here:
1. We want Grafana alerting, when proxying requests, to support SigV4 authentication. This would allow the existing UI to work against the prometheus-compatible endpoints in the AWS API (i.e. with this alone, grafana would treat AWS like a vanilla alertmanager, and be able to read but not write rules).
2. We want Grafana alerting to implement AWS's bespoke API for rule creation, so that Grafana can create rules in AMP.

Is my understanding correct here, did I assess these requests correctly, and do you also feel that this split makes sense? If so, you view one request as higher priority than another?

Secondly:
In order to detect which alerting implementation Grafana is talking to (cortex, prometheus, thanos, loki, etc) we have logic that effectively "sniffs" the API being proxied. This is necessary so that we can determine which endpoints are supported. Presumably, we'd need to do this for AWS as well, so that we know to use the right rule creation endpoint.

How might Grafana determine that the system it's talking to is actually AMP, in order to enable the right authentication or rule API plugin? Is there some signifier or base path in the URL perhaps, that would be guaranteed to work? Alternatively, is there an endpoint we can hit to obtain info about the target system, that would let Grafana know that it's talking to AMP as opposed to cortex or prometheus?

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/alerting-squad
